smartseq2_single_sample/hca-dcp/prepare_and_analyze_ss2_single_sample.wdl

```diff
@@ -0,0 +1,119 @@
+import "ss2_single_sample.wdl" as ss2
+
+task GetInputs {
+  String bundle_uuid
+  String bundle_version
+
+  command <<<
+    python <<CODE
+    import json
+    import requests
+    import subprocess
+
+    # Get bundle manifest
+    uuid = '${bundle_uuid}'
+    version = '${bundle_version}'
+    print('Getting bundle manifest for id {0}, version {1}'.format(uuid, version))
+
+    url = "https://hca-dss.czi.technology/v1/bundles/" + uuid + "?version=" + version + "&replica=gcp&directurls=true"
+    print('GET {0}'.format(url))
+    response = requests.get(url)
+    print('{0}'.format(response.status_code))
+    print('{0}'.format(response.text))
+    manifest = response.json()
+
+    bundle = manifest['bundle']
+    uuid_to_url = {}
+    name_to_meta = {}
+    for f in bundle['files']:
+        uuid_to_url[f['uuid']] = f['url']
+        name_to_meta[f['name']] = f
+    
+    print('Downloading assay.json')
```

Comment:
I expect this kind of wdl will need to be replicated across pipelines, do you think it would be worth it to extract these pieces of code into a package, with a `download_from_uuid` function?

I was separately wondering if we could convert the python portion of the secondary-analysis repo into something installable so I could reuse your cromwell commands. That would be a good place to put a function like the above, which would then be called twice more below to get the other files. 

Alternatively, you could inline `def download_from_uuid` to save on code and improve later maintainability if we ever need to change. I don't know if inline function defs are too meta. 😄 

I'd be happy to contribute to any of this.

Reply:
Good suggestions. I like the idea of extracting some of the Python from the WDL into a package. Made a ticket for it:
https://elastc.com/c/2BDXUEQ4/197-extract-wrapper-wdl-python-into-package